### PR TITLE
Fixes #913 Messages hexdump script

### LINKED ISSUE #913: [examples] Automatically generate hex dump of gnet messages
Script input should be a set of message objects , which are meant to be serialized (binary) and printed as an hex dump , similar to the snippet below . Beware of fields excluded from serialization (e.g. message context) .

```
02000000 ........................... Block version: 2

b6ff0b1b1680a2862a30ca44d34
6d9e8
910d334beb48ca0c0000000000000000 ... Hash of previous block's header
9d10aa52ee949386ca9385695f04ede2
70dda20810decd12bc9b048aaab31471 ... Merkle root

24d95a54 ........................... Unix time: 1415239972
30c31b18 ........................... Target: 0x1bc330 * 256**(0x18-3)
fe9f0864 ........................... Nonce
```

### PR DESCRIPTION
**Pull request by stdevEdu**

### CODE REVIEW

**Comment by gz-c** [comment]
Commands should go into the `cmd/` folder, in this case

`cmd/message-hexdump/message-hexdump.go`

**Comment by gz-c** [comment]
If this is just utility methods you could put them in a new file in the gnet library

**Comment by olemis** [comment]
@stdevEdu we do not need a new command for this. Testing examples should invoke hex dump code .

@gz-c hex dump code belongs in `testutil` package . @stdevEdu please change that accordingly .

File: extra/message_hexdump.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
+package extra
```
Comment (olemis) [inline extra/message_hexdump.go]: Package should be `testutil` . Please move this file to the corresponding location .

File: extra/message_hexdump.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	"reflect"
+	"github.com/skycoin/skycoin/src/daemon/gnet"
+	"github.com/skycoin/skycoin/src/cipher/encoder"
+	"strconv"
+)
+
+func serializeMessage(msg gnet.Message) []byte {
+	t := reflect.ValueOf(msg).Elem().Type()
+	msgID := gnet.MessageIDMap[t]
+	//if !succ {
```
Comment (olemis) [inline extra/message_hexdump.go]: Remove unnecessary comments .

Comment (olemis) [reply extra/message_hexdump.go]: Besides , [travis-ci build seems to be failing](https://travis-ci.org/skycoin/skycoin/builds/338800056?utm_source=github_status&utm_medium=notification)

**Comment by stdevEdu** [comment]
I'm unable to inherit PingMessage & PongMessage from gnet.Message.
Skipped tests on the for the moment

**Comment by olemis** [comment]
@stdevEdu I'll take a look into the issue relayed to Ping and Pong messages . For the moment please take a look into this error so that travis-ci checks will be back to green again 

```
$ gometalinter --disable-all -E goimports --tests --vendor ./...
src/testutil/message_hexdump.go:1::warning: file is not goimported (goimports)
src/testutil/message_hexdump_test.go:1::warning: file is not goimported (goimports)
The command "gometalinter --disable-all -E goimports --tests --vendor ./..." exited with 1.
```

**Comment by stdevEdu** [comment]
@olemis, I think this PR is ready to merge

**Comment by stdevEdu** [comment]
@olemis, travis-ci checks issues solved.

File: src/testutil/message_hexdump.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+	"reflect"
+	"strconv"
+	"time"
+
+	"github.com/skycoin/skycoin/src/cipher"
+	"github.com/skycoin/skycoin/src/cipher/encoder"
+	"github.com/skycoin/skycoin/src/daemon/gnet"
+)
+
+func serializeMessage(msg gnet.Message) []byte {
```
Comment (gz-c) [inline src/testutil/message_hexdump.go]: this is a copy of `encodeMessage` from gnet, should make `encodeMessage` a public method and use that

Comment (gz-c) [reply src/testutil/message_hexdump.go]: #970 exports `gnet.EncodeMessage`

Comment (olemis) [reply src/testutil/message_hexdump.go]: Very nice addition . This is something I needed for #735 and even implemented in 42fd42cf .

Comment (stdevEdu) [reply src/testutil/message_hexdump.go]: It is a copy, but when I try to use `gnet.EncodeMessage`, I get this error:

`panic: Attempted to serialize message struct not in MessageIdMap: &{1234 7890 5 <nil> false}`

That's why I remove the subsequent error verification in `serializeMessage`. But removing it in the original would spur further errors.

Comment (stdevEdu) [reply src/testutil/message_hexdump.go]: Also, there is still the problem of casting `PingMessage` and `PongMessage` as `gnet.Message` type

Comment (gz-c) [reply src/testutil/message_hexdump.go]: `panic: Attempted to serialize message struct not in MessageIdMap: &{1234 7890 5 <nil> false}` is the message not registered?

Comment (stdevEdu) [reply src/testutil/message_hexdump.go]: Problem solved

**Pull request closed without merging**